### DISCUSSION TITLE
[ListBox] How to retrigger "SelectedItemChanged" when clicking on an already selected item?

### DISCUSSION THREAD

**Discussion by YkTru** [dotnet/wpf#10075] [Q&A]
I made a cascading listboxes control:

https://github.com/user-attachments/assets/c2987ac5-db3e-4833-89ee-914f13be7238

---
**However**, if I click on an already selected item at any level, the SelectionChanged event does not fire again, and consequently, the Property Editor does not update either:

https://github.com/user-attachments/assets/f49e83bf-f365-497a-b581-338e3d793308

I know I can use "invoke" etc. though I'm using a MVVM-like solution (Elmish.WPF), so I guess a behavior + custom dependency property might be a solution?

Is there a way to create an "ItemClicked" event that listen to any selected/unselected item click at any level so that when its triggered, I could call SelectionChanged even on an already seleted Item (that would in fact reselect itself so the Property Editor updates)? (I hope I dont sound too confused here)

*Calling Deselect  then SelectItem seems a bad option IMO (I don't want stuff to "flash"), though I may be wrong

( here's the [repo](https://github.com/YkTru/Cascading-ListBoxes---SubModelSelectedItem/tree/F%23_version2) )

Thank you

**Comment by miloush** [comment]
How do you expect it to work with keyboard navigation, how would you "reselect" the item? Would using GotFocus to update the property editor work? Otherwise you can use MouseLeftButtonDown event on the ListBoxItem...
